launch.go

```diff
@@ -474,8 +474,18 @@ func launch(api screwdriver.API, buildID int, rootDir, emitterPath, metaSpace, s
 		return err, "", ""
 	}
 
-	// Always merge parent event meta if it exists (Issue #3234)
-	if event.ParentEventID != 0 { // If has parent event, fetch meta from parent event
+	// Note: event and parent event meta are mutually exclusive
+	// the first event in the build chain for a restart case will use the parent event meta
+	// on launcher exit, the meta will be updated to the current event meta
+
+	// merge event meta if available
+	if len(event.Meta) > 0 { // If has meta, marshal it
+		log.Printf("Fetching Event Meta JSON %v", event.ID)
+		if event.Meta != nil {
+			mergedMeta = deepMergeJSON(mergedMeta, event.Meta)
+		}
+	} else if event.ParentEventID != 0 { // otherwise, fetch from parent event
+		// If has parent event, fetch meta from parent event
```

Comment:
I believe the code is self-explanatory
```suggestion
	if len(event.Meta) > 0 {
		log.Printf("Fetching Event Meta JSON %v", event.ID)
		if event.Meta != nil {
			mergedMeta = deepMergeJSON(mergedMeta, event.Meta)
		}
	} else if event.ParentEventID != 0 {
```